### PR TITLE
Avoid workflows which write to the project directory

### PR DESCRIPTION
**Pull request by peterahrens**
This branch intends to avoid workflows that write to the `MatrixDepot.jl` project directory. There are two parts of MatrixDepot which have this behavior: The default data directory and the custom matrix generator directory. So far, I have only replaced the default data directory with a directory managed by Scratch.jl (#73). Soon, I will make a PR to this branch that replaces the custom matrix generators with a simpler package-based solution. This branch supersedes #74 and is intended to be merged at some point in the future when the MatrixDepot developers feel comfortable requiring Julia 1.5.

### CODE REVIEW

**Comment by peterahrens** [comment]
My proposed changes to replace custom code loading with a package-based solution are now ready for review in https://github.com/peterahrens/MatrixDepot.jl/pull/1.

**Comment by andreasnoack** [comment]
I think we should go ahead with this change. This shouldn't be a problem for anybody who is on an older version of Julia. If an important bugfix appears at some point then we can branch off from a commit before this one and backport the bugfix and make a bugfix release without these changes.

**Comment by peterahrens** [comment]
Shall I also merge in the proposed change in peterahrens#1 ?

**Comment by andreasnoack** [comment]
I'm in favor. It seems like the right solution to me. We should hear if @KlausC has any comments.

File: doc/user.rst [hunk 1]
```diff
@@ -75,12 +75,12 @@ That points to the default user directory which can be changed by an environment
 
     MATRIXDEPOT_USERDIR=/...
 
-The data directory is found by::
+The data directory is queried by::
 
     julia> MatrixDepot.data_dir()
-    "/home/.../.julia/dev/MatrixDepot/data"
+    "/home/.../.julia/scratchspaces/MatrixDepot/data"
```
Comment (KlausC) [inline doc/user.rst:81]: Please cross-check the dir name. I think to remember it contains a randomized part.

Comment (peterahrens) [reply doc/user.rst]: Good catch! I have updated the path to the correct UUID version.

**Review by KlausC — APPROVED** [review]

**Pull request merged**